### PR TITLE
Add support for the `SECURITY` hive type

### PR DESCRIPTION
**Pull request by ravenexp**
1. Adds a new hive type and updates the auto-detection routine.
1. Adds a binary SID decoding utility function.
1. Adds a sample plugin that operates on the newly supported `SECURITY` hive.

### CODE REVIEW

**Review by tincho9 — CHANGES_REQUESTED** [review]
Looks great! Can you please add a test?

**Comment by ravenexp** [comment]
Yes, I thought about doing that. The problem is I didn't have a suitable registry hive file I could add to the test data. I tested the code on the real Windows machine `SECURITY` hive, but I couldn't just upload it because it's full of security-sensitive info.

Perhaps you could suggest me a good way of creating such a test file, because I couldn't think of any. Even if I do a clean VM install, I can't join this machine to an AD domain because I don't have administrator permissions in any domains. And the `domain_sid` plugin depends on the machine being joined to a domain.

File: regipy/utils.py [hunk 1]
```diff
@@ -155,6 +158,43 @@ def try_decode_binary(data, as_json=False, max_len=MAX_LEN):
...
+    if num_words == 5:
+        # Machine SID + RID
+        components = unpack("<5L", data[8:28])
+        return "S-1-5-%d-%d-%d-%d-%d" % components
+    elif num_words == 4:
+        # Machine SID only
+        components = unpack("<4L", data[8:24])
+        return "S-1-5-%d-%d-%d-%d" % components
+    else:
+        raise ValueError("Unexpected SID component count")
```
Comment (tincho9) [inline regipy/utils.py:195]: Please inherit from `RegipyException`

**Review by tincho9 — APPROVED** [review]
Lets leave it without a test then

File: regipy/utils.py (hunk 1 above)
Comment (ravenexp) [reply regipy/utils.py]: I've added `NtSidDecodingException(RegipyException)`  class.

**Pull request merged**